# Análises Base CEPAGRI


### 1. Import das bases

In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
%matplotlib inline

### 2. Tratamento das bases

#### Criando array com o nome das planilhas com as informações

In [52]:
planilha = [str(1997+w) for w in range(0,20)]
planilha

['1997',
 '1998',
 '1999',
 '2000',
 '2001',
 '2002',
 '2003',
 '2004',
 '2005',
 '2006',
 '2007',
 '2008',
 '2009',
 '2010',
 '2011',
 '2012',
 '2013',
 '2014',
 '2015',
 '2016']

### Lendo cada planilha separando por abas iguais (111, 222, 333 e 265), em 4 dataframes

Se a coluna vier com a string "Unnamed" contido no nome, ela é dropada, pois é um erro de leitura de planilha do excel

In [53]:
data_111 = []
for p in planilha:
    name =p+'.xlsx'
    data_111.append(pd.read_excel(name, sheetname="111"))

In [54]:
for i in range(len(planilha)):
    to_drop = []
    for var in data_111[i].columns:
        if("Unnamed" in var):
            to_drop.append(var)
    data_111[i] = data_111[i].drop(to_drop, axis=1) 

In [55]:
data_222 = []
for p in planilha:
    name =p+'.xlsx'
    data_222.append(pd.read_excel(name, sheetname="222"))

In [56]:
for i in range(len(planilha)):
    to_drop = []
    for var in data_222[i].columns:
        if("Unnamed" in var):
            to_drop.append(var)
    data_222[i] = data_222[i].drop(to_drop, axis=1) 

#### Foi visto que nas abas 333, a partir de 2013 não existe essa aba nas planilhas e portanto limitou-se a leitura até 2012

In [59]:
data_333 = []
for p in planilha:
    if(int(p)<2013):
        name =p+'.xlsx'
        print(name)
        data_333.append(pd.read_excel(name, sheetname="333"))

1997.xlsx
1998.xlsx
1999.xlsx
2000.xlsx
2001.xlsx
2002.xlsx
2003.xlsx
2004.xlsx
2005.xlsx
2006.xlsx
2007.xlsx
2008.xlsx
2009.xlsx
2010.xlsx
2011.xlsx
2012.xlsx


In [61]:
for i in range(len(planilha)-4):
    to_drop = []
    for var in data_333[i].columns:
        if("Unnamed" in var):
            to_drop.append(var)
    data_333[i] = data_333[i].drop(to_drop, axis=1) 

In [62]:
data_265 = []
for p in planilha:
    name =p+'.xlsx'
    data_265.append(pd.read_excel(name, sheetname="265"))

In [63]:
for i in range(len(planilha)):
    to_drop = []
    for var in data_265[i].columns:
        if("Unnamed" in var):
            to_drop.append(var)
    data_265[i] = data_265[i].drop(to_drop, axis=1) 

#### Checando o tamanhos das bases

In [155]:
# for i in range(len(data_111)):
#     print("Base ", i, " tem tamanho ", data_111[i].shape, "\n")

In [156]:
# for i in range(len(data_222)):
#     print("Base ", i, " tem tamanho ", data_222[i].shape, "\n")

In [157]:
# for i in range(len(data_333)):
#     print("Base ", i, " tem tamanho ", data_333[i].shape, "\n")

In [158]:
# for i in range(len(data_265)):
#     print("Base ", i, " tem tamanho ", data_265[i].shape, "\n")

#### Consolidando as bases

In [71]:
full_111 = data_111[0]

for i in range(1,len(data_111) - 1):
    full_111 = pd.concat([full_111, data_111[i]], axis = 0)

In [74]:
full_111.shape

(997705, 18)

In [75]:
full_222 = data_222[0]

for i in range(1,len(data_222) - 1):
    full_222 = pd.concat([full_222, data_222[i]], axis = 0)

In [76]:
full_222.shape

(6548, 21)

In [77]:
full_333 = data_333[0]

for i in range(1,len(data_333) - 1):
    full_333 = pd.concat([full_333, data_333[i]], axis = 0)

In [78]:
full_333.shape

(25320, 10)

In [79]:
full_265 = data_265[0]

for i in range(1,len(data_265) - 1):
    full_265 = pd.concat([full_265, data_265[i]], axis = 0)

In [80]:
full_265.shape

(6498, 13)

#### Salvando as bases consolidadas em arquivo csv

In [81]:
full_111.to_csv("base_111.csv", index = False)
full_222.to_csv("base_222.csv", index = False)
full_333.to_csv("base_333.csv", index = False)
full_265.to_csv("base_265.csv", index = False)

### 3. Análise das bases

In [82]:
full_111.head(2)

Cod   Ano  Dia Juliano  Hora - minuto  \
0  111  1997          1.0             10   
1  111  1997          1.0             20   

   Veloc. Instantânea do Vento a 5 metros (m/s)  \
0                                       1618.00   
1                                          0.99   

   Veloc. Instantânea do Vento a 2 metros (m/s)  \
0                                      1115.000   
1                                         0.763   

   Direção do Vento no instante da aquisição (°)  \
0                                           8.32   
1                                          20.59   

   Radiação Incidente total de 10 leituras (KW/m2)  \
0                                              0.0   
1                                              0.0   

   Radiação Refletida Total de 10 leituras (KW/m2)  \
0                                              0.0   
1                                              0.0   

   Fluxo Positivo de calor no solo total de 10 leituras (W/m2)  \
0                                                0.0             
1                                                0.0             

   Fluxo negativo de calor no solo . total de 10 leituras (W/m2)  \
0                                             -443.0               
1                                             -443.0               

   Umidade relativa Média (%)  Temperatura do Ar Média (°C)  \
0                        92.3                         19.84   
1                        93.1                         19.63   

   Temperatura do solo  1 Média (°C) à 3cm no solo  \
0                                            21.04   
1                                            20.97   

   Temperatura do solo 2Média (°C) à 6cm no solo   \
0                                           22.68   
1                                           22.65   

  Temperatura do solo 3Média (°C) Colocado sensor de UV em 31/07/09  \
0                                              24.59                  
1                                              24.52                  

   Pressão (hPa) Chuva total do período (mm)  
0          943.0                           0  
1          943.0                           0

In [83]:
full_222.head(2)

Cod   Ano Dia Juliano Hora - minuto  \
0  222  1997           1          2400   
1  222  1997           2          2400   

   Veloc. máxima do Vento a 5 metros (m/s)  \
0                                   6031.0   
1                                   4756.0   

   Veloc. máxima do Vento a 2 metros (m/s)  \
0                                   2677.0   
1                                   1758.0   

   Direção do Vento no instante da veloc. Máx.(°)  \
0                                            99.6   
1                                           119.5   

  Horário de ocorrência vr máx. veloc. do vento  \
0                                          1923   
1                                            45   

   Radiação Incidente Total do dia (KW/m2)  \
0                                    270.8   
1                                    165.3   

   Radiação Refletida Total do dia (KW/m2)           ...            \
0                                    53.81           ...             
1                                    27.46           ...             

   Fluxo Negativo de calor no solo. Total do dia (KW/m2)  \
0                                             -34.76       
1                                             -26.86       

  Temperatura do Ar  Máxima do dia (°C)  \
0                                 30.16   
1                                 26.15   

  Horário de ocorrência da Temp. Ar Máxima(°C)  \
0                                         1255   
1                                         1417   

  Temperatura do Ar Mínima do dia (°C)  \
0                                18.50   
1                                19.15   

   Horário de ocorrência da Temp. Ar Minima  \
0                                       551   
1                                      2355   

  Temperatura do Solo 1 Máxima do dia (°C)  \
0                                    34.53   
1                                    27.66   

  Horário de ocorrência da Temp. Solo 1 Máxima.  \
0                                          1258   
1                                          1425   

  Temperatura do Solo 1 Mínima do dia (°C)  \
0                                    19.15   
1                                    19.65   

   Horário de ocorrência da Temp. Solo 1 minima Chuva total do dia (mm)  
0                                           606                     0.0  
1                                          2352                     0.0  

[2 rows x 21 columns]

In [84]:
full_333.head()

Cod   Ano Dia Juliano Hora - minuto  \
0  333  1997           1           700   
1  333  1997           1           900   
2  333  1997           1          1400   
3  333  1997           1          1500   
4  333  1997           1          2100   

   Veloc. Instantânea do Vento a 5 metros (m/s)  \
0                                        3344.0   
1                                        2618.0   
2                                        2422.0   
3                                        2089.0   
4                                        4501.0   

   Veloc. Instantânea do Vento a 2 metros (m/s)  \
0                                        1973.0   
1                                        1544.0   
2                                        1422.0   
3                                        1253.0   
4                                        2509.0   

   Direção do Vento no instantânea  (°)  Umidade relativa Instantânea  (%)  \
0                                 89.10                              89.70   
1                                185.60                              80.20   
2                                 44.37                              62.04   
3                                 37.73                              62.90   
4                                138.60                              82.60   

   Temperatura do Ar Instantânea (°C) Pressão  
0                               20.47     942  
1                               23.38     944  
2                               28.23     942  
3                               27.86     941  
4                               21.21     944

In [86]:
full_265.head()

Cod   Ano Dia Juliano Hora - minuto  Chuva das 7 h as 7 h  \
0  265  1997           1           700                   0.0   
1  265  1997           2           700                   0.0   
2  265  1997           3           700                   0.0   
3  265  1997           4           700                   0.0   
4  265  1997           5           700                   0.0   

   Temperatura do Ar máxima do dia °C  \
0                               31.11   
1                               30.16   
2                               26.15   
3                               25.74   
4                               29.69   

  Horário de ocorrência da Temp. Ar máxima  \
0                                     1356   
1                                     1255   
2                                     1417   
3                                     1208   
4                                     1625   

   Temperatura do Ar Minima do dia (°C)  \
0                                 18.50   
1                                 19.21   
2                                 17.88   
3                                 17.81   
4                                 17.95   

   Horário de ocorrência da Temp. Ar minima  Umidade máxima  Hora  \
0                                     551.0             NaN   NaN   
1                                     510.0             NaN   NaN   
2                                     533.0             NaN   NaN   
3                                     126.0             NaN   NaN   
4                                     535.0             NaN   NaN   

   Umidade mínima  Hora.1  
0             NaN     NaN  
1             NaN     NaN  
2             NaN     NaN  
3             NaN     NaN  
4             NaN     NaN

### 3.1. Analise para a base 111

In [87]:
full_111.columns

Index(['Cod', 'Ano', 'Dia Juliano', 'Hora - minuto',
       'Veloc. Instantânea do Vento a 5 metros (m/s)',
       'Veloc. Instantânea do Vento a 2 metros (m/s)',
       'Direção do Vento no instante da aquisição (°)',
       'Radiação Incidente total de 10 leituras (KW/m2)',
       'Radiação Refletida Total de 10 leituras (KW/m2)',
       'Fluxo Positivo de calor no solo total de 10 leituras (W/m2)',
       'Fluxo negativo de calor no solo . total de 10 leituras (W/m2)',
       'Umidade relativa Média (%)', 'Temperatura do Ar Média (°C)',
       'Temperatura do solo  1 Média (°C) à 3cm no solo',
       'Temperatura do solo 2Média (°C) à 6cm no solo ',
       'Temperatura do solo 3Média (°C) Colocado sensor de UV em 31/07/09',
       'Pressão (hPa)', 'Chuva total do período (mm)'],
      dtype='object')

In [161]:
full_111.Cod.unique()

array([111, 333, 265, 222], dtype=int64)

#### Aparentemente algo está errado em 2005, é só a marcação de código ou os dados também?

Dentro da base 111 em 2005 existe Cod que são 222, 333 e 265

In [103]:
print(full_111[full_111.Cod == 222]['Ano'].unique())
print(full_111[full_111.Cod == 333]['Ano'].unique())
print(full_111[full_111.Cod == 265]['Ano'].unique())

[2005]
[2005]
[2005]


### Checando valores missings

In [104]:
full_111.isnull().sum()

Cod                                                                     0
Ano                                                                     0
Dia Juliano                                                             0
Hora - minuto                                                           0
Veloc. Instantânea do Vento a 5 metros (m/s)                            0
Veloc. Instantânea do Vento a 2 metros (m/s)                            0
Direção do Vento no instante da aquisição (°)                           0
Radiação Incidente total de 10 leituras (KW/m2)                         0
Radiação Refletida Total de 10 leituras (KW/m2)                         0
Fluxo Positivo de calor no solo total de 10 leituras (W/m2)             0
Fluxo negativo de calor no solo . total de 10 leituras (W/m2)        1825
Umidade relativa Média (%)                                           1826
Temperatura do Ar Média (°C)                                         1826
Temperatura do solo  1 Média (°C) à 3c

Pequeno teste de tendência de temperatura, onde foi visto que existem valores fora das escalas esperadas

In [134]:
# temp_2001 = full_111[full_111.Ano == 2015]['Temperatura do Ar Média (°C)']

In [159]:
# plt.figure(figsize=(20,6))
# plt.plot(temp_2001[(temp_2001<50) & (temp_2001 >=0)])

### 3.2. Informações das variáveis:

In [154]:
full_111.describe()

Cod            Ano    Dia Juliano  Hora - minuto  \
count  997705.000000  997705.000000  997705.000000  997705.000000   
mean      111.503029    2006.049183     183.574166    1192.172977   
std        10.152882       5.476202     105.365608     692.751398   
min       111.000000    1997.000000     -53.360000      10.000000   
25%       111.000000    2001.000000      93.000000     610.000000   
50%       111.000000    2006.000000     184.000000    1210.000000   
75%       111.000000    2011.000000     275.000000    1810.000000   
max       333.000000    2139.000000     654.000000    6640.000000   

       Veloc. Instantânea do Vento a 5 metros (m/s)  \
count                                 997705.000000   
mean                                    1434.735636   
std                                     1332.101405   
min                                    -6999.000000   
25%                                        0.959000   
50%                                     1481.000000   
75%                                     2199.000000   
max                                     6996.000000   

       Veloc. Instantânea do Vento a 2 metros (m/s)  \
count                                 997705.000000   
mean                                     877.778706   
std                                     1114.843558   
min                                        0.000000   
25%                                        0.539000   
50%                                        1.010000   
75%                                     1696.000000   
max                                     6999.000000   

       Direção do Vento no instante da aquisição (°)  \
count                                  997705.000000   
mean                                      273.632008   
std                                       669.646808   
min                                         0.000000   
25%                                        93.100000   
50%                                       155.500000   
75%                                       259.400000   
max                                      6999.000000   

       Radiação Incidente total de 10 leituras (KW/m2)  \
count                                    997705.000000   
mean                                        791.007292   
std                                        1644.318924   
min                                       -6999.000000   
25%                                           0.000000   
50%                                           0.000000   
75%                                         471.800000   
max                                        6999.000000   

       Radiação Refletida Total de 10 leituras (KW/m2)  \
count                                    997705.000000   
mean                                        280.373982   
std                                         910.445572   
min                                           0.000000   
25%                                           0.000000   
50%                                           0.000000   
75%                                           0.919000   
max                                        6999.000000   

       Fluxo Positivo de calor no solo total de 10 leituras (W/m2)  \
count                                      997705.000000             
mean                                          375.408254             
std                                          1150.692398             
min                                             0.000000             
25%                                             0.000000             
50%                                             0.000000             
75%                                           217.000000             
max                                          6999.000000             

       Fluxo negativo de calor no solo . total de 10 leituras (W/m2)  \
count                                      995880.000000               
mean                                        -1383.991541               
std  

O que significa esses valores absurdos? 6999 e 0 6999 e demias valores totalmente fora de escala para a variável? é algum tipo de missing ou erro no equipamento?
